code/tests/btrfs_link_unlink.cpp

```diff
@@ -0,0 +1,144 @@
+/*
+Reproducing new btrfs bug reported
+
+1. Create directory A under TEST_MNT
+2. Create file foo in dir A
+3. Create a link bar to file foo in the same dir A
+4. Sync the changes
+5. Unlink bar
+6. Create file bar again
+7. fsync(bar)
+
+If a power fail occurs now, the file system must be mountable
+
+
+Fails on btrfs kernel 4.16. The filesystem becomes unmountable
+
+https://marc.info/?l=linux-btrfs&m=151983337811947&w=2
+https://marc.info/?l=linux-btrfs&m=151983349112005&w=2
+*/
+
+
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <string>
+#include <iostream>
+#include <dirent.h>
+#include <cstring>
+#include <errno.h>
+
+#include "BaseTestCase.h"
+#include "../user_tools/api/workload.h"
+#include "../user_tools/api/actions.h"
+#define TEST_FILE_FOO "foo"
+#define TEST_FILE_BAR "bar"
+#define TEST_DIR_A "test_dir_a"
+
+
+using fs_testing::tests::DataTestResult;
+using fs_testing::user_tools::api::WriteData;
+using fs_testing::user_tools::api::WriteDataMmap;
+using fs_testing::user_tools::api::Checkpoint;
+using std::string;
+
+#define TEST_FILE_PERMS  ((mode_t) (S_IRWXU | S_IRWXG | S_IRWXO))
+
+namespace fs_testing {
+namespace tests {
+
+
+class BtrfsLinkUnlink: public BaseTestCase {
+ public:
+  virtual int setup() override {
+
+    //initialize paths
+    foo_path = string(mnt_dir_) + "/"+ TEST_DIR_A + "/" + TEST_FILE_FOO;    
```

Comment:
`mnt_dir_` should be a string already, so you shouldn't have to make it into a string :)